File: guides/micronaut-dynamodb/java/src/main/java/example/micronaut/Book.java
package example.micronaut;

import io.micronaut.core.annotation.NonNull;
import io.micronaut.serde.annotation.Serdeable;

import jakarta.validation.constraints.NotBlank;

@Serdeable // <1>
public class Book implements Identified {

    @NonNull
    @NotBlank // <2>
    private final String id;

    @NonNull
    @NotBlank // <2>
    private final String isbn;

    @NonNull
    @NotBlank // <2>
    private final String name;

    public Book(@NonNull String id,
                @NonNull String isbn,
                @NonNull String name) {
        this.id = id;
        this.isbn = isbn;
        this.name = name;
    }

    @Override
    @NonNull
    public String getId() {
        return id;
    }

    @NonNull
    public String getIsbn() {
        return isbn;
    }

    @NonNull
    public String getName() {
        return name;
    }
}

File: guides/micronaut-microservices-distributed-tracing-jaeger/bookcatalogue/python/src/example/micronaut/book.py
from dataclasses import dataclass
from typing import Annotated

from jakarta.validation.constraints import NotBlank
from micronaut.serde.annotation import Serdeable


@Serdeable
@dataclass(frozen=True)
class Book:
    isbn: Annotated[str, NotBlank]
    name: Annotated[str, NotBlank]
